inst/csv/replacementPatterns.csv

```diff
@@ -492,6 +492,7 @@ impala,"CHAR\n+","CHAR(1)\n"
 impala,"CHAR)","CHAR(1))"
 impala,"CONSTRAINT @a DEFAULT NOW()",""
 impala,"DEFAULT NOW()",""
+impala,"DEFAULT 0",""

```

Comment:
What id there will be another default value? E.g. `DEFAULT 123`? Won't work

Reply:
I've seen this approach in a couple of other places too, which makes me worry a lot:

The idea around sql render is to give semantically equivalent statements between dialects, but if the translation goes from:

`CREATE TABLE someTable (int x NOT NULL DEFAULT 0);`

to:

`CREATE TABLE someTable (int x NOT NULL);`

That is definitely not semantically equivalent.  Normally the solution is to just work the limitation in the technical solution (don't depend on default values, for example).

We've sorta slipped by this one with index rules where not applying an index won't make a semantic difference to query logic.  But default values on columns?  There's a big difference between a table with a default value and one without.

Reply:
Although this is not the best solution, doesn't seem that we have another options and such cases should be handled by middle-tier

Reply:
I agree with @chrisknoll : removing the default may solve the short-term problem because now Impala doesn't throw an error anymore. But whoever wrote the default in their SQL in the first place will be expecting that default in their application, so we would get unexpected behavior down the line. Worst case is that we will get incorrect results without knowing they are incorrect. 

I would argue we should prefer a clear hard error message, which allows us to go back to the application developer and request it to be removed. The application developer can then assess the impact of removing the default, rather than just SqlRender blindly removing it.

I have started down the path of [giving warnings when people use SQL that may cause issues on some platforms](https://github.com/OHDSI/SqlRender/blob/master/java/org/ohdsi/sql/SqlTranslate.java#L444). Perhaps we could add a similar warning if people start using default values?

Reply:
@schuemie , I'd like to follow up on the issue. Are there any improvement suggestions/solutions from you or we could proceed?

Reply:
Hi @pavgra . In my last comment, I remarked that I think blindly removing defaults is a bad idea, and suggested that instead the need for defaults should be removed in the application you're developing. 

So if by 'proceed' you mean you are going to do that, then yes, we can proceed and I'll close this pull request.